src/main/java/org/broadinstitute/hellbender/tools/copynumber/arguments/CopyNumberArgumentValidationUtils.java

```diff
@@ -184,4 +187,82 @@ public static AnnotatedIntervalCollection validateAnnotatedIntervalsSubset(final
                 "Annotated intervals do not contain all specified intervals.");
         return new AnnotatedIntervalCollection(locatableCollection.getMetadata(), subsetAnnotatedIntervals);
     }
+
+    /**
+     * Validate that input files and/or directories are readable if they are not {@code null} (i.e., optional inputs).
+     */
+    public static void validateInputs(final File ... inputs) {
+        if (inputs != null) {
+            for (final File input : inputs) {
+                if (input != null) {
+                    if (input.isFile()) {
+                        IOUtils.canReadFile(input);
+                    } else if (input.isDirectory() && !input.canRead()) {
+                        throw new UserException.CouldNotReadInputFile(input);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Validate that output files are writeable, whether or not they already exist.
+     */
+    public static void validateOutputFiles(final File ... outputs) {
+        Utils.nonNull(outputs);
+        for (final File output : outputs) {
+            Utils.nonNull(output);
+            if ((output.exists() && !output.canWrite()) || (!output.exists() && !output.getAbsoluteFile().getParentFile().canWrite())) {
+                throw new UserException.CouldNotCreateOutputFile(output, ": The output file is not writeable.");
+            }
+        }
+    }
+
+    /**
+     * Validate that output directories are writeable.  If a directory does not exist, create it.
+     */
+    public static void validateAndPrepareOutputDirectories(final File ... outputs) {
```

Comment:
I think you should combine this with `validateOutputFiles()` since `File` can be either a file or directory. Otherwise it seems that one could pass in dirs/files to the wrong function and get unexpected results.

Reply:
Yeah, overall I'm not very happy with any of these validate input/output methods.  I originally had a single `validateOutputs` method, but the logic got a little ugly.  Let me try again.

In any case, we should probably overhaul all of the CNV Files over to Paths and revisit this issue.

Reply:
Actually, I think the reason I went with two methods is that we can't check whether a non-existing File is a file or a directory, and we want to perform different actions in either case (check for write permission of the parent directory of the file and create the directory, respectively).  So we have to rely on calling code to correctly make the choice for us.  I think that the method names should suffice to direct a developer to the correct method, but I can change the argument names to make it even more explicit.

Might be a way to do this better, but I'm OK to punt until the overhaul.  There are other behaviors that we could decide upon (such as whether or not we want to create intermediate directories, which this code doesn't do).  Baby steps.